src/modules/git_status.rs

```diff
@@ -230,43 +230,51 @@ struct RepoStatus {
 }
 
 impl RepoStatus {
-    fn is_conflicted(status: &str) -> bool {
-        status.starts_with("u ")
-    }
-
-    fn is_deleted(status: &str) -> bool {
+    fn is_deleted(short_status: &str) -> bool {
         // is_wt_deleted || is_index_deleted
-        status.starts_with("1 .D") || status.starts_with("1 D")
+        short_status.contains('D')
     }
 
-    fn is_renamed(status: &str) -> bool {
-        // is_wt_renamed || is_index_renamed
-        // Potentially a copy and not a rename
-        status.starts_with("2 ")
+    fn is_modified(short_status: &str) -> bool {
+        // is_wt_modified || is_wt_added
+        short_status.ends_with('M') || short_status.ends_with('A')
     }
 
-    fn is_modified(status: &str) -> bool {
-        // is_wt_modified
-        status.starts_with("1 .M") || status.starts_with("1 .A")
+    fn is_staged(short_status: &str) -> bool {
+        // is_index_modified || is_index_added
+        short_status.starts_with('M') || short_status.starts_with('A')
     }
 
-    fn is_staged(status: &str) -> bool {
-        // is_index_modified || is_index_new
-        status.starts_with("1 M") || status.starts_with("1 A")
-    }
+    fn parse_normal_status(&mut self, short_status: &str) {
+        if Self::is_deleted(short_status) {
+            self.deleted += 1;
+        }
+
+        if Self::is_modified(short_status) {
+            self.modified += 1;
+        }
 
-    fn is_untracked(status: &str) -> bool {
-        // is_wt_new
-        status.starts_with("? ")
+        if Self::is_staged(short_status) {
+            self.staged += 1;
+        }
     }
 
     fn add(&mut self, s: &str) {
-        self.conflicted += Self::is_conflicted(s) as usize;
-        self.deleted += Self::is_deleted(s) as usize;
-        self.renamed += Self::is_renamed(s) as usize;
-        self.modified += Self::is_modified(s) as usize;
-        self.staged += Self::is_staged(s) as usize;
-        self.untracked += Self::is_untracked(s) as usize;
+        if let Some(line_type) = s.chars().next() {
+            match line_type {
+                '1' => self.parse_normal_status(&s[2..4]),
+                '2' => {
+                    self.renamed += 1;
+                    self.parse_normal_status(&s[2..4])
+                }
+                'u' => self.conflicted += 1,
+                '?' => self.untracked += 1,
+                '!' => (),
+                _ => log::error!("Unknown line type in git status output"),
+            }
+        } else {
+            log::error!("Missing line type in git status output");
+        }
```

Comment:
Not a blocker but I think the if let can be removed from here
```suggestion
           match s.chars().next() {
              Some('1') => self.parse_normal_status(&s[2..4]),
              Some('2') => {
                  self.renamed += 1;
                  self.parse_normal_status(&s[2..4])
              }
              Some('u') => self.conflicted += 1,
              Some('?') => self.untracked += 1,
              Some('!') => (),
              Some(_) => log::error!("Unknown line type in git status output"),
              _ => log::error!("Missing line type in git status output"),
             }
```